### PR TITLE
Add explanation for multiple batch sizes to examples/README.md.

### PR DESCRIPTION
**Pull request by heiner**

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -47,3 +49,74 @@ Here's a sample result:
...
+    memory) is this partially treated as a sequence length as well.
+
+  * The virtual batch size, i.e., the number of samples in the `B`
+    dimension moolib consumes and adds to its gradient buffers before
+    a gradient descent step is happening. This can happen in two ways:
+    (1) A single peer could go through several backprop steps and keep
+    adding to its "running gradient" before it applies the gradient,
+    or (2) multiple peers go through one sample each and accumulate
+    their gradients. The virtual batch size is then
+    `number_of_peers * learner_batch_size`.
```
Comment (tscmoo) [inline examples/README.md:112]: It might be worth noting here that `number_of_peers` is not necessarily *all*  peers that participate in the training, but instead is just how many peers happened to participate in "this" gradient accumulation (and due to (1), a peer could participate multiple times, thus `number_of_peers` can even be higher than the total number of peers in the training).

Comment (heiner) [reply examples/README.md]: Good point, added ~this as a footnote.
